src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/casenotes/CaseNotesService.kt

```diff
@@ -121,11 +121,16 @@ class CaseNotesService(
     authorLastName = author.lastName,
     prisonId = agencyLocation?.id ?: offenderBooking.location?.id,
     caseNoteText = parseMainText(caseNoteText),
-    amendments = parseAmendments(caseNoteText),
+    amendments = parseAmendments(this),
     noteSourceCode = noteSourceCode,
     createdDatetime = LocalDateTime.of(dateCreation, timeCreation?.toLocalTime() ?: LocalTime.MIDNIGHT),
-    createdUsername = createdUserId,
+    createdUsername = createdUserId!!,
     auditModuleName = auditModuleName,
+    sourceSystem = if (auditModuleName in dpsModules && (modifiedUserId == null || modifiedUserId == createdUserId)) {
+      SourceSystem.DPS
+    } else {
+      SourceSystem.NOMIS
+    },
```

Comment:
logic specified in [the MICN ticket](https://dsdmoj.atlassian.net/browse/MICN-185)